src/pybind/mgr/dashboard/controllers/iscsi.py

```diff
@@ -1110,6 +1108,16 @@ def get_available_gateway():
                              component='iscsi')
 
 
+def get_active_portal(portal_names):
+    for portal_name in portal_names:
+        try:
+            IscsiClient.instance(gateway_name=portal_name).ping()
+            return portal_name
+        except (IscsiGatewayDoesNotExist, RequestException):
+            pass
+    return None
```

Comment:
From a load-balancing perspective, do we want to privilege always the first gateway? If not, an alternative approach could be shuffling the list items:
```suggestion
def get_active_portal(portal_names):
    for portal_name in random.shuffle(portal_names):
        try:
            IscsiClient.instance(gateway_name=portal_name).ping()
            return portal_name
        except (IscsiGatewayDoesNotExist, RequestException):
            pass
    return None
```

Additionally, given this call may involve up to `len(portal_names)` sequential HTTP requests, this method could become rather slow under unfavourable network condition or HTTP timeouts. A possibility would be: 
* to perform the pings in parallel and return as soon as the first response arrives,
* establish a short timeout (few seconds),
* and/or `ttl_cache` that result (so that every Dashboard API request doesn't trigger up to `len(portal_names)` iSCSI API back-end requests).

Reply:
Let's see what we can build in. At the moment I have simply rearranged the existing code. This ping check already existed in the code and is used in other essential functions (e.g. get_available_gateway or validate_rest_api), so nothing new. 

> do we want to privilege always the first gateway?

This is how the current code is doing it, e.g. `portals[0]['host']`. All code places that are now calling `get_active_portal` had previously used `foo[0]`. Shuffling the portals makes sense IMO.